generate.js

```diff
@@ -1,25 +1,56 @@
 // generate data as array of objects [{}, {}]
 // data fields -> id, name, url, name, rating, reviews, price, isPrime
-const faker = require('faker');
+const fs = require('fs');
 
+const AWS = 'https://s3-us-west-1.amazonaws.com/hamazon-product-images/';
 const RANDOM_BOOLEAN = [true, false];
-const DATA_SIZE = 10;
+const DATA_BASE = 1000000;
+const TIMES = 10;
+const COLS = ['id', 'name', 'url', 'rating', 'reviews', 'price', 'isPrime'];
 
-const generateData = (size) => {
-  const data = [];
-  for (let i = 1; i <= size; i++) {
-    const dataItem = {};
-    dataItem.id = i;
-    dataItem.name = `hamazon${i}`;
-    dataItem.url = faker.image.avatar();
-    dataItem.rating = (Math.random() * 5).toFixed(1);
-    dataItem.reviews = Math.floor(Math.random() * 500);
-    dataItem.price = (Math.random() * 1000).toFixed(2);
-    dataItem.isPrime = RANDOM_BOOLEAN[Math.floor(Math.random() * 2)];
-    data.push(dataItem);
+const ltrs = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j'];
+const generateName = (index, length) => {
+  let result = '';
+  for (let i = 0; i < length; i++) {
+    result += ltrs[index % 10];
+    index = Math.floor(index / 10);
   }
-  console.log(data);
-  return data;
+  return result;
 };
 
-generateData(DATA_SIZE);
+const createDataItem = (index) => {
+  const dataItem = {};
+  dataItem.id = index;
+  dataItem.name = generateName(index - 1, 7);
+  dataItem.url = `${AWS}${Math.floor(Math.random() * 10)}.jpeg`;
+  dataItem.rating = (Math.random() * 5).toFixed(1);
+  dataItem.reviews = Math.floor(Math.random() * 500);
+  dataItem.price = (Math.random() * 1000).toFixed(2);
+  dataItem.isPrime = RANDOM_BOOLEAN[Math.floor(Math.random() * 2)];
+  return dataItem;
+};
+
+const generateData = (start, end) => {
```

Comment:
I'm not sure about this, but I might consider just using a string instead of an array here, since the join operation wouldn't be necessary in that case. For very large amounts of data, having the join could potentially be more expensive than just successively appending the strings.